src/test/java/com/google/javascript/cl2dts/shouldResolveForwardDeclares/forward.js

```diff
@@ -0,0 +1,11 @@
+goog.provide('forward.A');
```

Comment:
My understanding is that `goog.forwardDeclare` is useful if your code mentions another type (`forward.D`), but when your end user compiles this into a JSBinary, the forwarded type may or may not be part of the compilation unit.

It's a hack that allows users to make sure they definitely don't compile some code in, and don't have to trust JSCompiler's code pruning to work (which it sometimes doesn't, due to code that has side effects on load).

I think an appropriate test would be to have a `goog.forwardDeclare('forward.D');`, but not have a `forward.D` on the compilation path at all.